### PR TITLE
simplify and fix optuna-hpo notebook

### PR DESCRIPTION
**Pull request by jameslamb**
Contributes to #402 .

I ran through `source/examples/rapids-optuna-hpo/notebook.ipynb` today. This proposes a few changes to that notebook based on that experience.

Added bonus... I tested this with Python 3.11 (for #377).

## Notes for Reviewers

I've left comments inline explaining the changes.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,10 @@
+*.bz2
 *.csv
 *.csv.gz
+*.csv.zip
+*.tar.gz
+*.zip
```
Comment (jameslamb) [inline .gitignore:6]: The `rapids-optuna-hpo` example ends up downloading a `.zip` file, which contains some inner zipped CSV files.

That's why I added `.zip` and `.csv.zip` here, and then though "might as well as some other common archives too", as I don't think we ever want any of these in source control.

File: source/examples/rapids-optuna-hpo/notebook.ipynb [hunk 2]
```diff
@@ -28,17 +28,17 @@
...
   {
    "cell_type": "code",
-   "execution_count": 1,
+   "execution_count": null,
    "metadata": {},
    "outputs": [],
    "source": [
     "## Run this cell to install optuna\n",
-    "# !pip install optuna"
+    "#!pip install optuna optuna-integration"
```
Comment (jameslamb) [inline source/examples/rapids-optuna-hpo/notebook.ipynb:36]: This notebook calls the following Python code:

```python
storage = optuna.integration.DaskStorage()
```

The `optuna.integration` module is provided by a separate package, `optuna-integration`, which is not pulled in by default by `pip install optuna`.

Adding this avoids a `ModuleNotFoundError` later on in the notebook.

File: source/examples/rapids-optuna-hpo/notebook.ipynb [hunk 3]
```diff
@@ -855,6 +318,22 @@
...
+    "\n",
+    "```text\n",
+    "[I 2024-08-06 09:41:40,161] Trial 1 finished with value: 8.238207899472073 and parameters: {'C': 40.573838784392514, 'penalty': 'l2', 'fit_intercept': True}. Best is trial 1 with value: 8.238207899472073.\n",
+    "... \n",
+    "[I 2024-08-06 09:41:58,423] Trial 143 finished with value: 8.210414278942531 and parameters: {'C': 0.3152731188939818, 'penalty': 'l1', 'fit_intercept': True}. Best is trial 52 with value: 8.205579602300705.\n",
+    "\n",
+    "Best params: {'C': 1.486491072441749, 'penalty': 'l2', 'fit_intercept': True}\n",
+    "Number of finished trials:  144\n",
+    "```"
+   ]
```
Comment (jameslamb) [inline source/examples/rapids-optuna-hpo/notebook.ipynb:335]: The logs generated above:

* are huge
* contain stacktraces with absolute paths from wherever the code was run

Proposing here to instead clear the output of that cell and just keep this shorter snippet of the logs in a markdown block.

**Review by jacobtomlinson — APPROVED** [review]
LGTM thanks @jameslamb

**Pull request merged**